crates/ra_assists/src/handlers/unwrap_block.rs

```diff
@@ -0,0 +1,412 @@
+use crate::{Assist, AssistCtx, AssistId};
+
+use ast::{BlockExpr, Expr, LoopBodyOwner};
+use ra_fmt::unwrap_trivial_block;
+use ra_syntax::{ast, AstNode, TextRange};
+
+// Assist: unwrap_block
+//
+// This assist removes if...else, for, while and loop control statements to just keep the body.
+//
+// ```
+// fn foo() {
+//     if true {<|>
+//         println!("foo");
+//     }
+// }
+// ```
+// ->
+// ```
+// fn foo() {
+//     println!("foo");
+// }
+// ```
+pub(crate) fn unwrap_block(ctx: AssistCtx) -> Option<Assist> {
+    let res = if let Some(if_expr) = ctx.find_node_at_offset::<ast::IfExpr>() {
```

Comment:
This will do a bunch of tree traversal. I think we should start the assist with `let l_curly = ctx.find_token_at_offset(T!['{']);` and than look at it's parent up to certain depth, rather than the full set of ancestors.

Reply:
In generally, it doesn't make sense to worry to much about assist perf, but it's good to not do premature pessimization if possible

Reply:
I updated with your advices. But I didn't add a `.take` method on ancestors iterator. Because it's a little bit subjective and I have no idea which limit we can set. You probably have a better idea so if you want to add this kind of limit let me know which limit I put.